release/RELEASE_PROCESS.md

```diff
@@ -2,48 +2,162 @@
 
 ## Create an issue for release tracking
 
-- Create an issue in [kserve/modelmesh-serving](https://github.com/kserve/modelmesh-serving)
+- Create an issue in [kserve/modelmesh-serving](https://github.com/kserve/modelmesh-serving/issues/new?title=release:)
 - Label the issue with `priority p0`
 - Label the issue with `kind process`
 
-## Releasing KServe components
+## Releasing ModelMesh components
 
-A release branch should be substantially _feature complete_ with respect to the intended release.
-Code that is committed to `main` may be merged or cherry-picked on to a release branch, but code that is directly committed to the release branch should be solely applicable to that release (and should not be committed back to main).
-In general, unless you're committing code that only applies to the release stream (for example, temporary hotfixes, backported security fixes, or image hashes), you should commit to `main` and then merge or cherry-pick to the release branch.
+A release branch should be substantially _feature complete_ with respect to the
+intended release. Code that is committed to `main` may be merged or cherry-picked
+on to a release branch, but code that is directly committed to the release branch
+should be solely applicable to that release (and should not be committed back to
+`main`). In general, unless you're committing code that only applies to the release
+stream (for example, temporary hotfixes, backported security fixes, or image hashes),
+you should commit to `main` and then merge or cherry-pick to the release branch.
 
 ## Create a release branch
 
-If you aren't already working on a release branch (of the form `release-${MAJOR}`, where `release-${MAJOR}` is a major-minor version number), then create one.
+Create a _release branch_ from `main` in the form of `release-${MAJOR}.${MINOR}`.
 Release branches serve several purposes:
 
-1.  They allow a release wrangler or other developers to focus on a release without interrupting development on `main`,
-1.  They allow developers to track the development of a release before a release candidate is declared,
-1.  They simplify back porting critical bug fixes to a patch level release for a particular release stream (e.g., producing a `v0.6.1` from `release-0.6`), when appropriate.
+1. They allow a release wrangler or other developers to focus on a release without interrupting development on `main`,
+2. They allow developers to track the development of a release before a release candidate is declared,
+3. They simplify back porting critical bug fixes to a patch level release for a particular release stream (e.g., producing a `v0.6.1` from `release-0.6`), when appropriate.
```

Comment:
nit: Any reason for changing the numbering here (same applies to below)?

Reply:
you mean we should use 0.10 as an example or how we utilize Mardown numbering here?